pandas/core/series.py

```diff
@@ -2517,6 +2518,18 @@ def _make_cat_accessor(self):
 
     cat = base.AccessorProperty(CategoricalAccessor, _make_cat_accessor)
 
+    def _dir_deletions(self):
+        rv = set()
+        if not is_datetimelike(self):
+            rv.add('dt')
+        if not com.is_categorical_dtype(self.dtype):
```

Comment:
all of these accessors (cat,dt,str) are mutually exclusive, so should set in a single routine. IOW, maybe change delete all, then add back in an if-then